pyocd/gdbserver/gdbserver.py

```diff
@@ -173,14 +176,20 @@ def __init__(self, session, core=None, server_listening_callback=None):
             semihost_console = semihost_io_handler
         self.semihost = semihost.SemihostAgent(self.target_context, io_handler=semihost_io_handler, console=semihost_console)
         
+        #
+        # If SWV is enabled, create a SWVReader thread. Note that we only do
+        # this if the core is 0: SWV is not a per-core construct, and can't
+        # be meaningfully read by multiple threads concurrently.
+        #
         self._swv_reader = None
-        if session.options.get("enable_swv"):
+
+        if session.options.get("enable_swv") and core == 0:
```

Comment:
There's a minor issue with limiting to core 0. The gdbserver subcommand accepts a `--core` argument so it only starts a gdbserver for the specified core(s). So if you only start up a gdbserver for core 1, SWV would not work with this check.

For now this change is fine to keep since it fixes the more common problem.

Probably the ultimate solution is to move the creation of all the console, semihosting, and SWV infrastructure to outside the `GDBServer` class.

Reply:
Yes, though the right fix is to have the `SWVReader` not have any notion of core whatsoever, and to handle core-based event handling at a different layer -- but I was trying to get this working with minimal change, not rearchitect it.

Reply:
Yes, exactly. And of course I don't expect you to rearchitect it! 😄